crates/accelerate/src/synthesis/permutation/utils.rs

```diff
@@ -144,3 +144,20 @@ pub fn decompose_cycles(cycles: &Vec<Vec<usize>>) -> Vec<(usize, usize)> {
 
     swaps
 }
+
+// create swaps
+//Need to fix the below implementation
+pub fn _create_swap_layer(pattern: &ArrayView1<usize>) -> Vec<(usize, usize)> {
+    // let mut permutation: Vec<usize> = pattern.iter().map(|&x| x as usize).collect();
```

Comment:
We need to call this function both for constructing an even layer (constructing conditional swaps starting with `0`) and an even layer (constructing conditional swaps starting with `1`), so you probably need to pass the `starting_point` (either `0` or `1`) as the argument.